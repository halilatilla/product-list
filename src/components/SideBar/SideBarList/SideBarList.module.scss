.title {
  font-weight: 500;
  text-transform: capitalize;
  margin-top: 24px;
  margin-bottom: 20px;
}

.item {
  color: var(--c-text-secondary);
  font-size: 14px;
  text-transform: capitalize;

  &:hover {
    color: var(--c-primary);
  }
}

.count {
  margin-left: 4px;
}

.active {
  color: var(--c-primary);
}
