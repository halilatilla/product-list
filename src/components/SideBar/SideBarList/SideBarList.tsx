import { FC } from 'react'
import classnames from 'classnames'

import { useAppSelector, useAppDispatch, setFilterByColor, setFilterByBrand, setOrderBy } from '@src/store'
import { getSortedListByAZ } from '@src/lib'
import { filterOptions } from '@src/constants'
import { Button } from '@src/components'
import { IFilterItem } from '@src/types'
import styles from './SideBarList.module.scss'

interface Props {
  className?: string
  title: string
  productValue: string
  items: IFilterItem[]
}

const SideBarList: FC<Props> = ({ className, items, title, productValue, ...rest }) => {
  const { filterByColor, filterByBrand, orderBy } = useAppSelector((state) => state.filter)
  const dispatch = useAppDispatch()

  const handleFilteredItem = ({ value, productValue }: { value: string; productValue: string }) => {
    if (productValue === filterOptions.ORDER) {
      if (orderBy === value) {
        dispatch(setOrderBy(''))
        return
      }

      dispatch(setOrderBy(value))
      return
    }

    if (productValue === filterOptions.COLOR) {
      if (filterByColor === value) {
        dispatch(setFilterByColor(''))
        return
      }

      dispatch(setFilterByColor(value))
      return
    }

    if (filterByBrand === value) {
      dispatch(setFilterByBrand(''))
      return
    }

    dispatch(setFilterByBrand(value))
  }

  return (
    <div className={styles.filter} {...rest}>
      {items.length > 0 && <p className={styles.title}>{title}</p>}
      <ul>
        {getSortedListByAZ(items, 'label').map(({ label, value, count }) => (
          <li key={label}>
            <Button
              className={classnames(styles.item, {
                [styles.active]: orderBy === value || filterByColor === value || filterByBrand === value,
              })}
              label={label}
              appearance="text"
              onClick={() => handleFilteredItem({ value, productValue })}
            >
              {count && <span className={styles.count}>({count})</span>}
            </Button>
          </li>
        ))}
      </ul>
    </div>
  )
}

export default SideBarList
