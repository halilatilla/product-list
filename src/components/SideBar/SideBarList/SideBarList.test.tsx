import { render } from '@src/test/test-utils'

import SideBarList from './SideBarList'

describe('SideBarList', () => {
  const PROPS = {
    products: [],
    className: '',
    title: '',
    productValue: '',
    items: [],
  }

  it.only('renders correctly', () => {
    const { container } = render(<SideBarList {...PROPS} />)
    expect(container).toMatchSnapshot()
  })
})
