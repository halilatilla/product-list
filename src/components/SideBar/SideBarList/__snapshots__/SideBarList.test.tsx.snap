// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideBarList renders correctly 1`] = `
<div>
  <div
    class="filter"
    products=""
  >
    <ul />
  </div>
</div>
`;
